module/zfs/dmu_objset.c

```diff
@@ -796,6 +797,9 @@ dmu_objset_create_impl(spa_t *spa, dsl_dataset_t *ds, blkptr_t *bp,
 	os->os_phys->os_type = type;
 	if (dmu_objset_userused_enabled(os)) {
 		os->os_phys->os_flags |= OBJSET_FLAG_USERACCOUNTING_COMPLETE;
+		if (dmu_objset_userobjused_enabled(os))
+			os->os_phys->os_flags |=
+			    OBJSET_FLAG_USEROBJACCOUNTING_COMPLETE;
```

Comment:
> The per-object accounting does not depend on the existing size accounting.

After reading through the patch more carefully let me clarify this.  This change should be able to operate independently of the the size accounting, but it appears that's not the case.  If someone creates a pool without size accounting and then enabled object accounting it won't work.  I think we should either.

* Automatically enable size accounting when object accounting is enabled, OR
* Prevent object accounting from being enabled if size accounting is not enabled.